packages/teleport/src/Discover/Shared/PingTeleportContext.tsx

```diff
@@ -30,27 +32,41 @@ export function PingTeleportProvider(props: {
 
   const { timedOut, result } = usePoll(
     signal =>
-      ctx.desktopService
-        .fetchDesktopServices(
-          ctx.storeUser.getClusterId(),
-          {
-            search: `${INTERNAL_RESOURCE_ID_LABEL_KEY} ${joinToken.internalResourceId}`,
-            limit: 1,
-          },
-          signal
-        )
-        .then(res => {
-          if (res.agents.length) {
-            return res.agents[0];
-          }
-
-          return null;
-        }),
+      servicesFetchFn(signal).then(res => {
+        if (res.agents.length) {
+          return res.agents[0];
+        }
+
+        return null;
+      }),
     timeout,
     active,
     props.interval
   );
 
+  function servicesFetchFn(signal: AbortSignal) {
+    const clusterId = ctx.storeUser.getClusterId();
+    const request = {
+      search: `${INTERNAL_RESOURCE_ID_LABEL_KEY} ${joinToken.internalResourceId}`,
+      limit: 1,
+    };
+    switch (props.serviceKind) {
+      case 'node':
```

Comment:
Can we make use of the shared `ResourceKind` enum here? https://github.com/gravitational/webapps/blob/5e80f7f88fce2628a88629c103f71da3da229fd0/packages/teleport/src/Discover/Shared/ResourceKind.ts#L17-L23